lib/Parse/ParseDecl.cpp

```diff
@@ -2005,13 +2005,23 @@ ParserStatus Parser::parseDecl(SmallVectorImpl<Decl*> &Entries,
                                                   false);
         }
       }
+      if (CurDeclContext) {
+        auto nominal =
+          CurDeclContext->getAsNominalTypeOrNominalTypeExtensionContext();
```

Comment:
This won't work for extensions. We're still in parsing; we haven't resolved any names yet. If you want to do names for extensions, you should try to get common patterns out of `getExtendedTypeLoc`; otherwise, you can just dyn_cast to NominalTypeDecl.